shell/platform/android/io/flutter/app/FlutterActivity.java

```diff
@@ -5,277 +5,132 @@
 package io.flutter.app;
 
 import android.app.Activity;
+import android.content.Context;
 import android.content.Intent;
-import android.os.Build;
+import android.content.res.Configuration;
 import android.os.Bundle;
-import android.view.Window;
-import android.view.WindowManager;
-import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.app.FlutterActivityDelegate.ViewFactory;
 import io.flutter.plugin.common.PluginRegistry;
 import io.flutter.plugin.common.PluginRegistry.Registrar;
-import io.flutter.plugin.platform.PlatformPlugin;
-import io.flutter.view.FlutterMain;
 import io.flutter.view.FlutterView;
 
-import java.util.ArrayList;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
-
 /**
  * Base class for activities that use Flutter.
  */
-public class FlutterActivity extends Activity implements PluginRegistry {
-    private final Map<String, Object> pluginMap = new LinkedHashMap<>(0);
-    private final List<RequestPermissionResultListener> requestPermissionResultListeners = new ArrayList<>(0);
-    private final List<ActivityResultListener> activityResultListeners = new ArrayList<>(0);
-    private final List<NewIntentListener> newIntentListeners = new ArrayList<>(0);
-    private final List<UserLeaveHintListener> userLeaveHintListeners = new ArrayList<>(0);
-    private FlutterView flutterView;
-
-    private String[] getArgsFromIntent(Intent intent) {
-        // Before adding more entries to this list, consider that arbitrary
-        // Android applications can generate intents with extra data and that
-        // there are many security-sensitive args in the binary.
-        ArrayList<String> args = new ArrayList<String>();
-        if (intent.getBooleanExtra("trace-startup", false)) {
-            args.add("--trace-startup");
-        }
-        if (intent.getBooleanExtra("start-paused", false)) {
-            args.add("--start-paused");
-        }
-        if (intent.getBooleanExtra("use-test-fonts", false)) {
-            args.add("--use-test-fonts");
-        }
-        if (intent.getBooleanExtra("enable-dart-profiling", false)) {
-            args.add("--enable-dart-profiling");
-        }
-        if (intent.getBooleanExtra("enable-software-rendering", false)) {
-            args.add("--enable-software-rendering");
-        }
-        if (!args.isEmpty()) {
-            String[] argsArray = new String[args.size()];
-            return args.toArray(argsArray);
-        }
-        return null;
-    }
+public class FlutterActivity extends Activity implements FlutterView.Provider, PluginRegistry, ViewFactory {
+    private final FlutterActivityDelegate delegate = new FlutterActivityDelegate(this, this);
```

Comment:
Would it make sense to declare this as the FlutterActivityEvents type?  

Right now it looks like there's nothing that enforces that FlutterActivityEvents contains all activity overrides that are forwarded by FlutterActivity to the delegate.

FlutterActivity would also need FlutterView.Provider and PluginRegistry aliases for the delegate to call methods defined through those interfaces.